packages/grafana-ui/src/components/Sparkline/utils.test.ts

```diff
@@ -0,0 +1,93 @@
+import { FieldSparkline, FieldType } from '@grafana/data';
+
+import { preparePlotFrame } from './utils';
+
+describe('Prepare Sparkline plot frame', () => {
+  it('should return sorted array if x-axis numeric', () => {
+    const sparkline: FieldSparkline = {
+      x: {
+        name: 'x',
+        values: [5, 4, 3, 2, 1],
+        type: FieldType.number,
+        config: {},
+      },
+      y: {
+        name: 'y',
+        values: [1, 2, 3, 4, 5],
+        type: FieldType.number,
+        config: {},
+      },
+    };
+
+    const frame = preparePlotFrame(sparkline);
+
+    expect(frame.fields[0].values).toEqual([1, 2, 3, 4, 5]);
+    expect(frame.fields[1].values).toEqual([5, 4, 3, 2, 1]);
```

Comment:
The y-values need to remain in relation with the x-values. So if the x-values get sorted, the y-values also need to follow the same sort so that some y-value still remains related to a specific time value, for example.

Reply:
dah 😅